src/test/java/uk/gov/pay/products/service/ProductApiTokenManagerTest.java

```diff
@@ -0,0 +1,182 @@
+package uk.gov.pay.products.service;
+
+import ch.qos.logback.classic.Level;
+import ch.qos.logback.classic.Logger;
+import ch.qos.logback.classic.spi.ILoggingEvent;
+import ch.qos.logback.classic.spi.LoggingEvent;
+import ch.qos.logback.core.Appender;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.slf4j.LoggerFactory;
+import uk.gov.pay.commons.model.TokenPaymentType;
+import uk.gov.pay.products.client.publicauth.model.CreateApiTokenRequest;
+import uk.gov.pay.products.client.publicauth.model.NewApiTokenFromPublicAuthResponse;
+import uk.gov.pay.products.client.publicauth.model.TokenSource;
+import uk.gov.pay.products.config.ProductsConfiguration;
+import uk.gov.pay.products.exception.FailToReplaceApiTokenException;
+import uk.gov.pay.products.exception.MissingApiTokenException;
+import uk.gov.pay.products.fixtures.ProductEntityFixture;
+import uk.gov.pay.products.model.Product;
+import uk.gov.pay.products.util.ProductType;
+
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.Entity;
+import javax.ws.rs.client.Invocation;
+import javax.ws.rs.client.WebTarget;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.util.Optional;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static uk.gov.pay.products.service.ProductApiTokenManager.NEW_API_TOKEN_PATH;
+
+@ExtendWith(MockitoExtension.class)
+public class ProductApiTokenManagerTest {
+    private static final String PUBLICAUTH_URL = "https://publicauth.url";
+    private static final String EMAIL_ADDRESS_FOR_REPLACING_API_TOKENS = "pay-products@gov.uk";
+    private static final String NEW_API_TOKEN = "New API Token";
```

Comment:
Rather using the same new token, maybe we could have different one for each test? e.g. “New API token for payment link”, “New API token for demo payment” etc. Just to make what’s going on a little clearer.

Reply:
This has been done.